clar2wasm/src/wasm_generator.rs

```diff
@@ -260,708 +263,308 @@ impl WasmGenerator {
-                            args.get(2).unwrap_or(&DEFAULT_EXPR),
+                            args.get_name(0)?,
+                            args.get_expr(1)?,
+                            args.get_expr(2)?,
                         ),
                         DefineFunctions::PersistedVariable => self.traverse_define_data_var(
                             builder,
                             expr,
-                            args.get(0)
-                                .unwrap_or(&DEFAULT_EXPR)
-                                .match_atom()
-                                .unwrap_or(&DEFAULT_NAME),
-                            args.get(1).unwrap_or(&DEFAULT_EXPR),
-                            args.get(2).unwrap_or(&DEFAULT_EXPR),
-                        ),
-                        DefineFunctions::Trait => {
-                            let params = if !args.is_empty() { &args[1..] } else { &[] };
-                            self.traverse_define_trait(
-                                builder,
-                                expr,
-                                args.get(0)
-                                    .unwrap_or(&DEFAULT_EXPR)
-                                    .match_atom()
-                                    .unwrap_or(&DEFAULT_NAME),
-                                params,
-                            )
-                        }
-                        DefineFunctions::UseTrait => self.traverse_use_trait(
-                            builder,
-                            expr,
-                            args.get(0)
-                                .unwrap_or(&DEFAULT_EXPR)
-                                .match_atom()
-                                .unwrap_or(&DEFAULT_NAME),
-                            args.get(1)
-                                .unwrap_or(&DEFAULT_EXPR)
-                                .match_field()
-                                .unwrap_or(&TraitIdentifier {
-                                    contract_identifier: QualifiedContractIdentifier::transient(),
-                                    name: DEFAULT_NAME.clone(),
-                                }),
-                        ),
-                        DefineFunctions::ImplTrait => self.traverse_impl_trait(
-                            builder,
-                            expr,
-                            args.get(0)
-                                .unwrap_or(&DEFAULT_EXPR)
-                                .match_field()
-                                .unwrap_or(&TraitIdentifier {
-                                    contract_identifier: QualifiedContractIdentifier::transient(),
-                                    name: DEFAULT_NAME.clone(),
-                                }),
+                            args.get_name(0)?,
+                            args.get_expr(1)?,
+                            args.get_expr(2)?,
                         ),
+                        _ => Ok(()),
```

Comment:
Similarly here, I'd prefer to keep this as an exhaustive list so it's easy to track what's left to implement. Cyle made this improvement to one of my previous PRs and I think it's a good idea.

Reply:
request for further motivation above